tracer/src/Datadog.Trace/LibDatadog/TraceExporter.cs

```diff
@@ -92,4 +116,21 @@ protected override bool ReleaseHandle()
 
         return true;
     }
+
+    private unsafe int GetBodyLen(IntPtr body)
+    {
+        if (body == IntPtr.Zero)
+        {
+            throw new ArgumentNullException(nameof(body));
+        }
```

Comment:
In hindsight, throwing is expensive, we should maybe just do a fast return here (and maybe add a debug log given that this is unexpected), and then in the calling method check that len > 0, and if not, exit gracefully. My concern is that this is called a lot, so if there _is_ any issue here, we're likely going to start writing a _lot_ of logs...

Reply:
this will also affect inlining. Let's avoid throwing in an hot path, if this is absolutely necessary then let's create a different NonInlined function and throw from there.